pkg/provider/onboardbase/client/client.go

```diff
@@ -0,0 +1,349 @@
+	Title string `json:"title,omitempty"`
+	Id string `json:"id,omitempty"`
+}
+
+type secretResponseBodyData struct {
+	Project secretResponseBodyObject `json:"project,omitempty"`
+	Environment secretResponseBodyObject `json:"environment,omitempty"`
+	Team secretResponseBodyObject `json:"team,omitempty"`
+	Secrets []string `json:"secrets,omitempty"`
+}
+
+type secretResponseBody struct {
+	Data secretResponseBodyData `json:"data,omitempty"`
+	Message string `json:"message,omitempty"`
+	Status string `json:"status,omitempty"`
+}
+
+type SecretResponse struct {
+	Name  string
+	Value string
+}
+
+type SecretsResponse struct {
+	Secrets  Secrets
+	Body     []byte
+}
+
+func NewOnboardbaseClient(onboardbaseAPIKey, onboardbasePasscode string) (*OnboardbaseClient, error) {
+
+
+	tlsConfig := &tls.Config{
+		MinVersion: tls.VersionTLS12,
+	}
+	httpTransport := &http.Transport{
+		DisableKeepAlives: true,
+		TLSClientConfig:   tlsConfig,
+	}
+	client := &OnboardbaseClient{
+		OnboardbaseAPIKey: onboardbaseAPIKey,
+		OnboardbasePassCode: onboardbasePasscode,
+		VerifyTLS:    true,
+		UserAgent:    "onboardbase-external-secrets",
+		httpClient: &http.Client{
+			Timeout: 10 * time.Second,
+			Transport: httpTransport,
+		},
+	}
+
+
+	if err := client.SetBaseURL("https://public.onboardbase.com/api/v1/"); err != nil {
+		return nil, &APIError{Err: err, Message: "setting base URL failed"}
+	}
+
+	return client, nil
+}
+
+func (c *OnboardbaseClient) BaseURL() *url.URL {
+	u := *c.baseURL
+	return &u
+}
+
+func (c *OnboardbaseClient) SetBaseURL(urlStr string) error {
+	baseURL, err := url.Parse(strings.TrimSuffix(urlStr, "/"))
+
+	if err != nil {
+		return err
+	}
+
+	if baseURL.Scheme == "" {
+		baseURL.Scheme = "https"
+	}
+
+	c.baseURL = baseURL
+	return nil
+}
+
+func (c *OnboardbaseClient) Authenticate() error {
+
+	if _, err := c.performRequest("/team/members", "GET", headers{}, queryParams{	}, httpRequestBody{}); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (c *OnboardbaseClient) getSecretsFromPayload(data secretResponseBodyData) (map[string]string, error) {
+	kv := make(map[string]string)
+	for _, secret := range data.Secrets {
+		passphrase := c.OnboardbasePassCode
+		decrypted, err := aesdecrypt.Run(secret, passphrase)
+		if err != nil {
+			return nil, &APIError{Err: err, Message: "unable to decrypt secret payload", Data: secret}
+		}
+		var decryptedJSON RawSecret
+		if err := json.Unmarshal([]byte(decrypted), &decryptedJSON); err != nil {
+		    return nil, &APIError{Err: err, Message: "unable to unmarshal secret payload", Data: decrypted}
+	    }
+		kv[decryptedJSON.Key] = decryptedJSON.Value
+	}
+	return kv, nil
+}
+
+func (c *OnboardbaseClient) GetSecret(request SecretRequest) (*SecretResponse, error) {
+	params := request.buildQueryParams()
+
+	response, err := c.performRequest("/secrets", "GET", headers{}, params, httpRequestBody{})
+	if err != nil {
+		return nil, err
+	}
+
+	var data secretResponseBody
```

Comment:
This structure is also not compatible with the expected payload generated afterwards

Reply:
This should now be legible